c/tree.c

```diff
@@ -851,10 +1021,14 @@ LOCAL MMDBW_status insert_record_into_current_record(
             );
         current_record->value.key = key;
     } else if (new_record->type == MMDBW_RECORD_TYPE_FIXED_NODE ||
+               new_record->type == MMDBW_RECORD_TYPE_FIXED_EMPTY ||
                new_record->type == MMDBW_RECORD_TYPE_NODE ||
                new_record->type == MMDBW_RECORD_TYPE_ALIAS) {
+        // We're inserting these. Overwrite if it was there.
         current_record->value.node = new_record->value.node;
     }
+    // TODO(wstorey@maxmind.com): Should we have an else? What about other
+    // record types? EMPTY isn't dealt with here.
```

Comment:
Actually, I think we should remove `MMDBW_RECORD_TYPE_FIXED_EMPTY` above. `->value` is not set on empty nodes.

Reply:
Ah good point. I was wondering about whether we should zero it in the case of EMPTY mainly. Although I don't know that we even get new_record->type being empty here actually... But we would FIXED_EMPTY of course.